front/components/assistant/TryAssistant.tsx

```diff
@@ -211,7 +218,22 @@ export function usePreviewAssistant({
     }, animationLength);
   };
 
+  useEffect(() => {
+    if (!isEqual(previousBuilderState.current, builderState)) {
+      setHasChanged(true);
+      previousBuilderState.current = builderState;
+    }
+  }, [builderState]);
+
   const submit = useCallback(async () => {
+    if (!isPreviewOpened) {
+      // Preview is not opened, no need to submit
+      return;
+    }
+    if (draftAssistant && !hasChanged) {
+      // No changes since the last submission
+      return;
+    }
     const a = await submitAssistantBuilderForm({
```

Comment:
Do we need to disable the preview while this is happening ? I'm wondering what happens when you just opened the drawer and your thing is updating ?

Reply:
You did not read the OP 😛

Reply:
Basically at first open and until there's a first draft saved it's a big loader. Then the refresh behavior is the same as before with the quick "highlight" that we have when the drawer is updated for a new version 🙇🏻‍♀️